### PR TITLE
soc: arm: nxp_imx: rt: enable SOC fixed MPU regions, add SDRAM0 region

### PR DESCRIPTION
**Pull request by danieldegrasse**
Enable SOC fixed MPU regions by default for the RT10xx/RT11xx SOC lines.

Additionally, add code to handle defining the SDRAM0 region as
device type (non cacheable, non shareable), when no SDRAM0 node is
present. This will resolve an issue present on the RT11xx series where
the core may execute speculative prefetches to the SDRAM region when
no SDRAM is present on the board, resulting in the system faulting.

This PR also adds documentation to the RT series boards to highlight the external memories present. This is intended to
improve the customer experience when booting a custom board with Zephyr

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

### CODE REVIEW

File: soc/arm/nxp_imx/rt/CMakeLists.txt [hunk 1]
```diff
@@ -9,9 +9,9 @@ zephyr_sources_ifdef(CONFIG_SOC_SERIES_IMX_RT10XX soc_rt10xx.c)
 zephyr_linker_sources_ifdef(CONFIG_NXP_IMX_RT_BOOT_HEADER
   ROM_START SORT_KEY 0 boot_header.ld)
 
-# Enable custom OCRAM noncacheable region
+# Add custom mpu regions
+zephyr_sources(mpu_regions.c)
```
Comment (decsny) [inline soc/arm/nxp_imx/rt/CMakeLists.txt:13]: maybe something should go in the release notes about this change?

Comment (danieldegrasse) [reply soc/arm/nxp_imx/rt/CMakeLists.txt:13]: Sure- I'll highlight this in the next update of this PR

File: soc/arm/nxp_imx/rt/mpu_regions.c [hunk 2]
```diff
@@ -1,37 +1,37 @@
...
-#endif
+
+#ifndef CONFIG_NXP_IMX_EXTERNAL_SDRAM
+	/*
+	 * Region 2 - mark SDRAM0 as device type memory to prevent core
+	 * from executing speculative prefetches against this region when
+	 * no SDRAM is present.
+	 */
+	MPU_REGION_ENTRY("SDRAM0",
+			SDRAM_BASE_ADDR, REGION_IO_ATTR(REGION_512M)),
```
[the remark this answers is not in the document]
Comment (danieldegrasse) [reply soc/arm/nxp_imx/rt/mpu_regions.c:30]: Yes, this is intended to override the default attributes assigned to the 512M region starting at `0x8000_0000` in the ARMv7-M system memory map: 
image

Indentation has been fixed in latest push

**Review by decsny — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Comment by danieldegrasse** [comment]
> @pabigot, since you did look at this PR, can you provide an approval?

Think you might have meant @pdgendt :)

**Review by pdgendt — APPROVED** [review]

**Pull request merged**